mkdocs/docs/how-to-release.md

```diff
@@ -243,3 +252,13 @@ Make sure to create a PR to update the [GitHub issues template](https://github.c
 ## Update the integration tests
 
 Ensure to update the `PYICEBERG_VERSION` in the [Dockerfile](https://github.com/apache/iceberg-python/blob/main/dev/Dockerfile).
+
+## Create a Github Release Note
+
+First, follow the directions for Setting the tag again to create a tag for the approved release version (e.g. `0.7.0`).
```

Comment:
I think it's possible to auto-create a tag with a release note

![Screenshot 2024-07-30 at 5 41 43 PM](https://github.com/user-attachments/assets/a4bb8fab-90c2-41b0-bf3a-48de5564c4d3)